File: src/main/java/io/github/liyze09/arms/network/packet/serverbound/LoginAcknowledged.java
package io.github.liyze09.arms.network.packet.serverbound;

import io.github.liyze09.arms.network.Connection;
import io.github.liyze09.arms.network.packet.ServerBoundPacketDecoder;
import io.netty.buffer.ByteBuf;
import org.jetbrains.annotations.NotNull;

public class LoginAcknowledged implements ServerBoundPacketDecoder {
    @Override
    public void decode(ByteBuf buf, @NotNull Connection connection) {
        connection.updateStatus(Connection.Status.CONFIGURATION);
    }
}

File: src/main/kotlin/io/github/liyze09/arms/network/packet/serverbound/LoginAcknowledged.kt
package io.github.liyze09.arms.network.packet.serverbound

import io.github.liyze09.arms.network.Connection
import io.github.liyze09.arms.network.packet.clientbound.ClientBoundKnownPacks
import io.netty.buffer.ByteBuf

object LoginAcknowledged : io.github.liyze09.arms.network.packet.ServerBoundPacketDecoder {
    override fun decode(buf: ByteBuf, connection: Connection) {
        connection.updateStatus(Connection.Status.CONFIGURATION)
        connection.sendPacket(null, io.github.liyze09.arms.network.packet.clientbound.FeatureFlags)
        connection.sendPacket(null, ClientBoundKnownPacks)
    }
}
